src/connectors/solana/solana.controller.ts

```diff
@@ -176,4 +251,450 @@ export class SolanaController {
+    } catch (error) {
+      console.error('Error confirming transaction:', error.message);
+      throw new Error(`Failed to confirm transaction: ${error.message}`);
+    }
+  }
+
+  public async confirmTransactionByAddress(
+    address: string,
+    signature: string,
+    connection: Connection,
+    timeout: number = 3000,
+  ): Promise<boolean> {
+    try {
+      const confirmationPromise = new Promise<boolean>(async (resolve, reject) => {
+        const payload = {
+          jsonrpc: '2.0',
+          id: 1,
+          method: 'getSignaturesForAddress',
+          params: [
+            address,
+            {
+              limit: 100, // Adjust the limit as needed
+              until: signature,
+            },
+          ],
+        };
+
+        const response = await fetch(connection.rpcEndpoint, {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+          body: JSON.stringify(payload),
+        });
+
+        if (!response.ok) {
+          reject(new Error(`HTTP error! status: ${response.status}`));
+          return;
+        }
+
+        const data = await response.json();
+
+        if (data.result) {
+          const transactionInfo = data.result.find((entry) => entry.signature === signature);
+
+          if (!transactionInfo) {
+            resolve(false);
+            return;
+          }
+
+          if (transactionInfo.err !== null) {
+            reject(
+              new Error(`Transaction failed with error: ${JSON.stringify(transactionInfo.err)}`),
+            );
+            return;
+          }
+
+          const isConfirmed =
+            transactionInfo.confirmationStatus === 'confirmed' ||
+            transactionInfo.confirmationStatus === 'finalized';
+          resolve(isConfirmed);
+        } else {
+          resolve(false);
+        }
+      });
+
+      const timeoutPromise = new Promise<boolean>((_, reject) =>
+        setTimeout(() => reject(new Error('Confirmation timed out')), timeout),
+      );
+
+      return await Promise.race([confirmationPromise, timeoutPromise]);
+    } catch (error) {
+      console.error('Error confirming transaction using signatures:', error.message);
+      throw new Error(`Failed to confirm transaction using signatures: ${error.message}`);
+    }
+  }
+
+  async sendAndConfirmTransaction(tx: Transaction, signers: Signer[] = []): Promise<string> {
+    const priorityFeesEstimate = await this.fetchEstimatePriorityFees(
+      this.connectionPool.getNextConnection().rpcEndpoint,
+    );
+
+    const validFeeLevels = ['min', 'low', 'medium', 'high', 'veryHigh', 'unsafeMax'];
+    const priorityFeeLevel = process.env.PRIORITY_FEE_LEVEL || 'medium';
+
+    // Ensure the priorityFeeLevel is valid, otherwise default to 'high'
+    const selectedPriorityFee = validFeeLevels.includes(priorityFeeLevel)
+      ? priorityFeesEstimate[priorityFeeLevel]
+      : priorityFeesEstimate.high;
```

Comment:
should this be medium to match line 469?

Reply:
Thanks for the catch!